base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -90,14 +91,21 @@ fn verify_timelocks(tx: &Transaction, current_height: u64) -> Result<(), Validat
 
 // This function checks that the inputs and outputs do not exist in the STxO set.
 fn verify_not_stxos<B: BlockchainBackend>(tx: &Transaction, db: &B) -> Result<(), ValidationError> {
-    // TODO: Implement tip header.
-    // let tip = db.fetch_tip_header()?;
-    let tip = db.fetch_last_header()?;
-
-    let BlockAccumulatedData { deleted, .. } = db.fetch_block_accumulated_data(&tip.hash())?;
+    // `ChainMetadata::best_block` must always have the hash of the tip block.
+    // NOTE: the backend makes no guarantee that the tip header has a corresponding full body (interrupted header sync,
+    // pruned node) however the chain metadata MUST always be correct
+    let metadata = db.fetch_chain_metadata()?;
+    let data = db
+        .fetch_block_accumulated_data(metadata.best_block())?
+        .unwrap_or_else(|| {
+            panic!(
```

Comment:
Panic seems a bit harsh here

Reply:
I'm in two minds about this. It _should_ be impossible for the best block in chain metadata to ever not be an existing block and have accumulated data. If that is not the case, something has gone wrong (the db can be considered "corrupt"). Thoughts? (happy to change this)